fendermint/actors/blobs/src/caller.rs

```diff
@@ -0,0 +1,743 @@
+                account.credit_free += credit;
+                account.gas_allowance += value;
+            }
+            Self::Sponsored(delegation) => {
+                delegation.from_account.credit_free += credit;
+                delegation.from_account.gas_allowance += value;
+            }
+        }
+
+        debug!("added {} credits to {}", credit, self.subscriber_address());
+        debug!(
+            "added {} gas fee allowance to {}",
+            value,
+            self.subscriber_address()
+        );
+    }
+
+    /// Returns the credit allowance for the subscriber.
+    #[allow(dead_code)]
+    pub fn credit_allowance(&self, current_epoch: ChainEpoch) -> CreditAllowance {
+        match self {
+            Self::Default((_, account)) => CreditAllowance {
+                amount: account.credit_free.clone(),
+                ..Default::default()
+            },
+            Self::Sponsored(delegation) => delegation.credit_allowance(current_epoch),
+        }
+    }
+
+    /// Returns the gas allowance for the subscriber.
+    pub fn gas_allowance(&self, current_epoch: ChainEpoch) -> GasAllowance {
+        match self {
+            Self::Default((_, account)) => GasAllowance {
+                amount: account.gas_allowance.clone(),
+                ..Default::default()
+            },
+            Self::Sponsored(delegation) => delegation.gas_allowance(current_epoch),
+        }
+    }
+
+    /// Commits new capacity for the subscriber.
+    pub fn commit_capacity(
+        &mut self,
+        size: u64,
+        cost: &Credit,
+        current_epoch: ChainEpoch,
+    ) -> anyhow::Result<(), ActorError> {
+        // Check subscriber's free credit
+        if &self.subscriber().credit_free < cost {
+            return Err(ActorError::insufficient_funds(format!(
+                "account {} has insufficient credit (available: {}; required: {})",
+                self.subscriber_address(),
+                &self.subscriber().credit_free,
+                cost
+            )));
+        }
+        match self {
+            Self::Default((_, account)) => {
+                account.capacity_used += size;
+                account.credit_committed += cost;
+                account.credit_free -= cost;
+            }
+            Self::Sponsored(delegation) => {
+                delegation.use_credit_allowance(cost, current_epoch)?;
+                delegation.from_account.capacity_used += size;
+                delegation.from_account.credit_committed += cost;
+                delegation.from_account.credit_free -= cost;
+            }
+        }
+
+        debug!("used {} bytes from {}", size, self.subscriber_address());
+        debug!(
+            "committed {} credits from {}",
+            cost,
+            self.subscriber_address()
+        );
+
+        Ok(())
+    }
+
+    /// Uncommits capacity for the subscriber.
+    pub fn uncommit_capacity(&mut self, size: u64, cost: &Credit) {
```

Comment:
nit but maybe "release" instead of "uncommit"?

Reply:
nice, make sense!